lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -189,25 +189,52 @@ static Value buildUnitNormalCdf(OpBuilder &b, Location &loc, Value x) {
   return buildNormalCdf(b, loc, x, zero, one);
 }
 
+template<typename TorchOpTy, typename MathOpTy>
+static Value createCalculationForMathOpWithDtypeConversion(
+              OpBuilder &b, Location loc, TypeConverter *converter,
+              ValueRange payloadArgs, Operation *op) {
+  auto opCasted = dyn_cast<TorchOpTy>(op);
+  Type dtype = converter->convertType(opCasted.getType())
+                    .template cast<RankedTensorType>()
+                    .getElementType();
```

Comment:
You can do

https://github.com/llvm/torch-mlir/blob/ae4714aced1c9416f3854530333601d42c309140/lib/Conversion/TorchToLinalg/Uncategorized.cpp#L915-L917

This allows you to get rid of the first template argument `TorchOpTy`